include/bluetooth/conn.h

```diff
@@ -297,6 +297,10 @@ struct bt_conn_le_info {
 	/* Connection maximum single fragment parameters */
 	const struct bt_conn_le_data_len_info *data_len;
 #endif /* defined(CONFIG_BT_USER_DATA_LEN_UPDATE) */
+#if defined(CONFIG_BT_SMP)
+	/** Whether or not the connection is bonded */
+	bool bonded;
```

Comment:
I feel a bit conflicted about this. Being bonded describes the relationship between two Bluetooth devices (whether they both store a common LTK). It's not really a property of a connection. Any reason why we can't have an API that takes as input `bt_addr_le_t` instead of `bt_conn`? I'm also worried that people not technically familiar with Bluetooth might confuse this to mean "encrypted" or "authenticated" when we associate the property with a connection instead of a peer address.

Reply:
Also, most existing APIs that we have related to bonding refer to an address and not a connection, e.g.:
```c
int bt_unpair(uint8_t id, const bt_addr_le_t *addr);
```

```c
/** Information about a bond with a remote device. */
struct bt_bond_info {
	/** Address of the remote device. */
	bt_addr_le_t addr;
};
void bt_foreach_bond(uint8_t id, void (*func)(const struct bt_bond_info *info,
					   void *user_data),
		     void *user_data);
```

From the `auth_cb` struct:
```c
void (*bond_deleted)(uint8_t id, const bt_addr_le_t *peer);
```

Reply:
I understand. The typical usecase I see is that a user wants to know if a `conn` from the `connected` or `disconnected` callbacks are bonded, hence why I went with `conn` as the input for this. 

We already have an internal function `bt_addr_le_is_bonded` which does exactly what you suggest, and we can simply make that public as an alternative. 

To use the `bt_addr_le_is_bonded` (or to otherwise use the address), the user will also need to obtain the conn ID and address (using `bt_conn_get_info`) and know that the address to use is the `info->le.dst` instead of the `info->le.src`, the `info->le.local`, or `info->le.remote` address values. 

So using the address to obtain the bonding information will require 2 function calls, and the user will need to know which of the 4 addresses to use. 

The advantage of using the address, rather than the connection pointer, is of course that the address method can be used (more reliably) outside of a connection. 

Another alternative I can suggest is adding a `bt_conn_is_bonded(bt_conn *conn)` as an inline function that calls `bt_conn_get_info` and then calls `bt_addr_le_is_bonded` with the ID and `dst`. That way we have 1) a proper `is_bonded` function that works on addresses, instead of conn pointers, as well as a shorthand way of using a conn pointer to obtain the address and ID to get the bonding state.

Reply:
I'd be ok with making `bt_addr_le_is_bonded()` public. We also have a public `bt_conn_get_dst()` function which fits here better than `bt_conn_get_info()`. I'm not completely against a wrapper function either, but I don't really like `bt_conn_is_bonded()` since it's not the connection that's bonded. Something like `bt_conn_peer_is_bonded()` might be better. Or am I making this unnecessarily complicated? :) I'm just afraid that when we start using a security related property with a connection people might associate it with the connection being secure, when that's not necessarily the case.

Reply:
I completely follow your concerns. We will still need to use `bt_conn_get_info` to get the `id` needed to to do `bt_addr_le_is_bonded` (or `bt_unpair`) for that matter, unless we implement a `bt_conn_get_id()` as well. 

That will however also mean that both `bt_addr_le_is_bonded` and `bt_unpair` are implicitly linked to a `bt_conn`, because `bt_conn_get_info` is the only way to get the ID, even if 99% of the time `BT_ID_DEFAULT` would likely work, but for connection agnostic implementations (such as the LE Audio services), we should not rely on `BT_ID_DEFAULT` but rather ensure that the ID is always correct.